File: Problem/Easy/1701-1800/1779. Find Nearest Point That Has the Same X or Y Coordinate/_b.java
// dont use it tho

// To Code : 30 min
// Execution : 2 ms

// Time complexity: O(n)
// Space complexity: O(1)

class Solution {
	public int nearestValidPoint(int x, int y, int[][] points) {
		int res = -1;
		int min_dis = Integer.MAX_VALUE;

		for (int i = 0; i < points.length; i++) {
			// if (x == points[i][0] || y == points[i][1]) {
			// since any one will become 0
			// after sub .. if its a valid point
			// ie. same
			if ((x - points[i][0]) * (y - points[i][1]) == 0) {
				int dis = Math.abs(x - points[i][0]) + Math.abs(y - points[i][1]);

				if (dis < min_dis) {
					min_dis = dis;
					res = i;
				}
			}
		}

		return res;
	}
}

File: Problem/Easy/1701-1800/1779. Find Nearest Point That Has the Same X or Y Coordinate/b.py
import math


class Solution:
    # def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
    def nearestValidPoint(self, x: int, y: int, points) -> int:
        res, smallest = -1, math.inf

        for i, (r, c) in enumerate(points):
            if (x == r or y == c) and ((dis := abs(x - r + y - c)) < smallest):
                smallest = dis
                res = i

        return res


# x = Solution().nearestValidPoint(
#     3, 4, [[1, 2], [3, 1], [2, 4], [2, 3], [4, 4]])
# print(x)
